sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -426,10 +458,17 @@ export namespace ConnectionContext {
           connectionError || contextError,
           "batching"
         );
+      }
+
+      // Calling onDetached on session receivers
+      const numSessionReceivers = getNumberOfReceivers(connectionContext, "session");
+      if (!state.wasConnectionCloseCalled && numSessionReceivers) {
```

Comment:
At this point should we just have the `state.wasConnectionCloseCalled` as a top level check (and then do nothing if it's true). I think it'd make the code a bit easier to understand.

Also, I believe the comment for our early-exit would be "no need to handle the close here, since the actual ConnectionCOntext.close() will close all the senders and receivers".